File: scala/src/main/scala/io/swagger/client/api/CloudCacheApi.scala
package io.swagger.client.api

import io.swagger.client.model.KeyBody
import io.swagger.client.model.CacheRequest
import io.swagger.client.model.Cache
import io.swagger.client.model.CacheItem
import io.swagger.client.ApiInvoker
import io.swagger.client.ApiException

import com.sun.jersey.multipart.FormDataMultiPart
import com.sun.jersey.multipart.file.FileDataBodyPart

import javax.ws.rs.core.MediaType

import java.io.File
import java.util.Date

import scala.collection.mutable.HashMap

class CloudCacheApi(val defBasePath: String = "http://api.cloudboost.io/",
                        defApiInvoker: ApiInvoker = ApiInvoker) {
  var basePath = defBasePath
  var apiInvoker = defApiInvoker

  def addHeader(key: String, value: String) = apiInvoker.defaultHeaders += key -> value 

  
  /**
   * clear all items from a cloud cache
   * clear all items from a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param body master key of app 
   * @return List[CacheRequest]
   */
  def clearCache (appId: String, cacheName: String, body: KeyBody) : Option[List[CacheRequest]] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/clear".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "DELETE", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "array", classOf[CacheRequest]).asInstanceOf[List[CacheRequest]])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * create a cloud cache
   * create a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param body master key of app 
   * @return Cache
   */
  def createCache (appId: String, cacheName: String, body: KeyBody) : Option[Cache] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/create".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[Cache]).asInstanceOf[Cache])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * delete all CloudCaches
   * delete all CloudCaches
   * @param appId Id of CloudApp 
   * @param body master key of app 
   * @return List[CacheRequest]
   */
  def deleteAllCache (appId: String, body: KeyBody) : Option[List[CacheRequest]] = {
    // create path and map variables
    val path = "cache/{app_id}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "DELETE", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "array", classOf[CacheRequest]).asInstanceOf[List[CacheRequest]])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * delete a CloudCache
   * delete a CloudCache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param body master key of app 
   * @return CacheRequest
   */
  def deleteCache (appId: String, cacheName: String, body: KeyBody) : Option[CacheRequest] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "DELETE", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[CacheRequest]).asInstanceOf[CacheRequest])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * delete item from a cloud cache
   * delete item from a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param dataKey key under which data to delete is cached 
   * @param body master key of app 
   * @return String
   */
  def deleteItem (appId: String, cacheName: String, dataKey: String, body: KeyBody) : Option[String] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/item/{data_key}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    .replaceAll("\\{" + "data_key" + "\\}",apiInvoker.escape(dataKey))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "DELETE", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[String]).asInstanceOf[String])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * get all CloudCaches
   * get all CloudCaches
   * @param appId Id of CloudApp 
   * @param body master key of app 
   * @return List[Cache]
   */
  def getAllCache (appId: String, body: KeyBody) : Option[List[Cache]] = {
    // create path and map variables
    val path = "cache/{app_id}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "array", classOf[Cache]).asInstanceOf[List[Cache]])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * get all items from a cloud cache
   * get all items from a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param body master key of app 
   * @return List[CacheItem]
   */
  def getAllItems (appId: String, cacheName: String, body: KeyBody) : Option[List[CacheItem]] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/items".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "array", classOf[CacheItem]).asInstanceOf[List[CacheItem]])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * get info about a CloudCache
   * get info about a CloudCache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param body master key of app 
   * @return Cache
   */
  def getCacheInfo (appId: String, cacheName: String, body: KeyBody) : Option[Cache] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[Cache]).asInstanceOf[Cache])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * get item from a cloud cache
   * get item from a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param dataKey key under which data should be got in cache 
   * @param body master key of app 
   * @return String
   */
  def getItem (appId: String, cacheName: String, dataKey: String, body: KeyBody) : Option[String] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/{data_key}/item".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    .replaceAll("\\{" + "data_key" + "\\}",apiInvoker.escape(dataKey))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[String]).asInstanceOf[String])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * count items in a cloud cache
   * count items in a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param body master key of app 
   * @return Integer
   */
  def getItemsCount (appId: String, cacheName: String, body: KeyBody) : Option[Integer] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/items/count".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[Integer]).asInstanceOf[Integer])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
  /**
   * add item to a cloud cache
   * add item to a cloud cache
   * @param appId Id of CloudApp 
   * @param cacheName Name of cache 
   * @param dataKey key under which data should be store in cache 
   * @param body cache data and master key included got from this object 
   * @return String
   */
  def set (appId: String, cacheName: String, dataKey: String, body: CacheRequest) : Option[String] = {
    // create path and map variables
    val path = "cache/{app_id}/{cache_name}/{data_key}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "app_id" + "\\}",apiInvoker.escape(appId))

    .replaceAll("\\{" + "cache_name" + "\\}",apiInvoker.escape(cacheName))

    .replaceAll("\\{" + "data_key" + "\\}",apiInvoker.escape(dataKey))

    

    val contentTypes = List("application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

    

    
    
    

    var postBody: AnyRef = body

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
      
    }

    try {
      apiInvoker.invokeApi(basePath, path, "PUT", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[String]).asInstanceOf[String])
         
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  
}

File: typescript-angular/API/Client/CloudCacheApi.ts
/// <reference path="api.d.ts" />

/* tslint:disable:no-unused-variable member-ordering */

namespace API.Client {
    'use strict';

    export class CloudCacheApi {
        protected basePath = 'http://api.cloudboost.io/';
        public defaultHeaders : any = {};

        static $inject: string[] = ['$http', '$httpParamSerializer'];

        constructor(protected $http: ng.IHttpService, protected $httpParamSerializer?: (d: any) => any, basePath?: string) {
            if (basePath) {
                this.basePath = basePath;
            }
        }

        private extendObj<T1,T2>(objA: T1, objB: T2) {
            for(let key in objB){
                if(objB.hasOwnProperty(key)){
                    objA[key] = objB[key];
                }
            }
            return <T1&T2>objA;
        }

        /**
         * clear all items from a cloud cache
         * clear all items from a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param body master key of app
         */
        public clearCache (appId: string, cacheName: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<Array<CacheRequest>> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/clear'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling clearCache');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling clearCache');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling clearCache');
            }
            let httpRequestParams: any = {
                method: 'DELETE',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * create a cloud cache
         * create a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param body master key of app
         */
        public createCache (appId: string, cacheName: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<Cache> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/create'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling createCache');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling createCache');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling createCache');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * delete all CloudCaches
         * delete all CloudCaches
         * @param appId Id of CloudApp
         * @param body master key of app
         */
        public deleteAllCache (appId: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<Array<CacheRequest>> {
            const localVarPath = this.basePath + 'cache/{app_id}'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling deleteAllCache');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling deleteAllCache');
            }
            let httpRequestParams: any = {
                method: 'DELETE',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * delete a CloudCache
         * delete a CloudCache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param body master key of app
         */
        public deleteCache (appId: string, cacheName: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CacheRequest> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling deleteCache');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling deleteCache');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling deleteCache');
            }
            let httpRequestParams: any = {
                method: 'DELETE',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * delete item from a cloud cache
         * delete item from a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param dataKey key under which data to delete is cached
         * @param body master key of app
         */
        public deleteItem (appId: string, cacheName: string, dataKey: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<string> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/item/{data_key}'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName))
                .replace('{' + 'data_key' + '}', String(dataKey));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling deleteItem');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling deleteItem');
            }
            // verify required parameter 'dataKey' is set
            if (!dataKey) {
                throw new Error('Missing required parameter dataKey when calling deleteItem');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling deleteItem');
            }
            let httpRequestParams: any = {
                method: 'DELETE',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * get all CloudCaches
         * get all CloudCaches
         * @param appId Id of CloudApp
         * @param body master key of app
         */
        public getAllCache (appId: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<Array<Cache>> {
            const localVarPath = this.basePath + 'cache/{app_id}'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling getAllCache');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling getAllCache');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * get all items from a cloud cache
         * get all items from a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param body master key of app
         */
        public getAllItems (appId: string, cacheName: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<Array<CacheItem>> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/items'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling getAllItems');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling getAllItems');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling getAllItems');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * get info about a CloudCache
         * get info about a CloudCache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param body master key of app
         */
        public getCacheInfo (appId: string, cacheName: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<Cache> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling getCacheInfo');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling getCacheInfo');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling getCacheInfo');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * get item from a cloud cache
         * get item from a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param dataKey key under which data should be got in cache
         * @param body master key of app
         */
        public getItem (appId: string, cacheName: string, dataKey: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<string> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/{data_key}/item'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName))
                .replace('{' + 'data_key' + '}', String(dataKey));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling getItem');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling getItem');
            }
            // verify required parameter 'dataKey' is set
            if (!dataKey) {
                throw new Error('Missing required parameter dataKey when calling getItem');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling getItem');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * count items in a cloud cache
         * count items in a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param body master key of app
         */
        public getItemsCount (appId: string, cacheName: string, body: KeyBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<number> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/items/count'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling getItemsCount');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling getItemsCount');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling getItemsCount');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * add item to a cloud cache
         * add item to a cloud cache
         * @param appId Id of CloudApp
         * @param cacheName Name of cache
         * @param dataKey key under which data should be store in cache
         * @param body cache data and master key included got from this object
         */
        public set (appId: string, cacheName: string, dataKey: string, body: CacheRequest, extraHttpRequestParams?: any ) : ng.IHttpPromise<string> {
            const localVarPath = this.basePath + 'cache/{app_id}/{cache_name}/{data_key}'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'cache_name' + '}', String(cacheName))
                .replace('{' + 'data_key' + '}', String(dataKey));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling set');
            }
            // verify required parameter 'cacheName' is set
            if (!cacheName) {
                throw new Error('Missing required parameter cacheName when calling set');
            }
            // verify required parameter 'dataKey' is set
            if (!dataKey) {
                throw new Error('Missing required parameter dataKey when calling set');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling set');
            }
            let httpRequestParams: any = {
                method: 'PUT',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
    }
}
